src/Engine/ProtoCore/DSASM/Heap.cs

```diff
@@ -263,6 +332,71 @@ public StackValue AllocatePointer(int size)
                     new MetaData { type = (int)PrimitiveType.kTypePointer });
         }
 
+        /// <summary>
+        /// Allocate a string, the string will be put in string table.
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns></returns>
+        private int AllocateStringInternal(string str)
+        {
+            int index;
+            if (!stringTable.TryGetPointer(str, out index))
+            {
+                index = AllocateInternal(Enumerable.Empty<StackValue>());
+                stringTable.AddString(index, str);
+                heapElements[index].MetaData = new MetaData { type = (int)PrimitiveType.kTypeString};
+            }
+            return index;
+        }
+
+        /// <summary>
+        /// Allocate string constant. String constant won't be garbage collected.
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns></returns>
+        public StackValue AllocateFixedString(string str)
```

Comment:
Just to clarify:
I see that both AllocateFixedString and AllocateString are public and AllocateString  will eventually call AllocateFixedString. What will be the reason I call one as opposed to the other?